common/persistence/cassandraVisibilityPersistence.go

```diff
@@ -446,6 +454,35 @@ func (v *cassandraVisibilityPersistence) ListClosedWorkflowExecutionsByStatus(
 	return response, nil
 }
 
+func (v *cassandraVisibilityPersistence) GetClosedWorkflowExecution(
+	request *GetClosedWorkflowExecutionRequest) (*GetClosedWorkflowExecutionResponse, error) {
+	execution := request.Execution
+	query := v.session.Query(templateGetClosedWorkflowExecution,
+		request.DomainUUID,
+		domainPartition,
+		execution.GetWorkflowId(),
+		execution.GetRunId())
+
+	iter := query.Iter()
+	if iter == nil {
+		return nil, &workflow.InternalServiceError{
+			Message: "GetClosedWorkflowExecution operation failed.  Not able to create query iterator.",
+		}
+	}
+
+	wfexecution, has := readClosedWorkflowExecutionRecord(iter)
```

Comment:
Make sure to call iter.Close(), this should return back an error which we could wrap into InternalServerError and return back to application.